### PR TITLE
XDG stable: use mock methods instead of notification lists

### PR DESCRIPTION
**Pull request by wmww**
Simplifies and standardizes how events are handled in XDG types

### CODE REVIEW

File: tests/wlr_foreign_toplevel_management_v1.cpp [hunk 1]
```diff
@@ -467,10 +467,9 @@ TEST_F(ForeignToplevelHandleTest, gets_activated)
 TEST_F(ForeignToplevelHandleTest, can_maximize_foreign)
 {
     wlcs::XdgToplevelStable::State state{0, 0, nullptr};
-    xdg_toplevel.add_configure_notification(
-        [&state](int32_t width, int32_t height, struct wl_array *states)
+    ON_CALL(xdg_toplevel, configure).WillByDefault([&state](auto... args)
         {
-            state = wlcs::XdgToplevelStable::State{width, height, states};
+            state = wlcs::XdgToplevelStable::State{args...};
```
Comment (RAOF) [inline tests/wlr_foreign_toplevel_management_v1.cpp:472]: Nice use of generic lambdas :)

File: tests/wlr_foreign_toplevel_management_v1.cpp [hunk 2]
```diff
@@ -895,21 +885,15 @@ TEST_F(ForeignToplevelHandleTest, can_activate_foreign)
 
 TEST_F(ForeignToplevelHandleTest, can_close_foreign)
 {
-    bool should_close{false};
-    xdg_toplevel.add_close_notification(
-        [&should_close]()
-        {
-            should_close = true;
-        });
+    EXPECT_CALL(xdg_toplevel, close()).Times(0);
```
Comment (RAOF) [inline tests/wlr_foreign_toplevel_management_v1.cpp:888]: I'm not *sure* that this does what you want? GMock expectations are additive; how does this `EXPECT_CALL().Times(0)` interact with the previous `EXPECT_CALL().Times(AnyNumber())`?

**Review by RAOF — CHANGES_REQUESTED** [review]
Yeah, I broadly think this is a much better approach.

Please remove the catch-all expectations, and I'll be most pleased to land this :grin:

File: tests/wlr_foreign_toplevel_management_v1.cpp (hunk 2 above)
Comment (wmww) [reply tests/wlr_foreign_toplevel_management_v1.cpp:888]: I believe new more specific `EXPECT_CALL`'s supersede old more general ones. If you remove the `EXPECT_CALL(xdg_toplevel, close()).Times(1);` a few lines down the test fails.

**Pull request merged**